File: clojure/src/spoonacular_api/specs/get_recipe_information_bulk_200_response_inner.clj
(ns spoonacular-api.specs.get-recipe-information-bulk-200-response-inner
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.get-recipe-information-200-response-extended-ingredients-inner :refer :all]
            [spoonacular-api.specs.get-recipe-information-200-response-wine-pairing :refer :all]
            )
  (:import (java.io File)))


(def get-recipe-information-bulk-200-response-inner-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :image) string?
   (ds/req :imageType) string?
   (ds/req :servings) float?
   (ds/req :readyInMinutes) int?
   (ds/req :license) string?
   (ds/req :sourceName) string?
   (ds/req :sourceUrl) string?
   (ds/req :spoonacularSourceUrl) string?
   (ds/req :aggregateLikes) int?
   (ds/req :healthScore) float?
   (ds/req :spoonacularScore) float?
   (ds/req :pricePerServing) float?
   (ds/req :analyzedInstructions) (s/coll-of string?)
   (ds/req :cheap) boolean?
   (ds/req :creditsText) string?
   (ds/req :cuisines) (s/coll-of string?)
   (ds/req :dairyFree) boolean?
   (ds/req :diets) (s/coll-of string?)
   (ds/req :gaps) string?
   (ds/req :glutenFree) boolean?
   (ds/req :instructions) string?
   (ds/req :ketogenic) boolean?
   (ds/req :lowFodmap) boolean?
   (ds/req :occasions) (s/coll-of string?)
   (ds/req :sustainable) boolean?
   (ds/req :vegan) boolean?
   (ds/req :vegetarian) boolean?
   (ds/req :veryHealthy) boolean?
   (ds/req :veryPopular) boolean?
   (ds/req :whole30) boolean?
   (ds/req :weightWatcherSmartPoints) float?
   (ds/req :dishTypes) (s/coll-of string?)
   (ds/req :extendedIngredients) (s/coll-of get-recipe-information-200-response-extended-ingredients-inner-spec)
   (ds/req :summary) string?
   (ds/req :winePairing) get-recipe-information-200-response-wine-pairing-spec
   })

(def get-recipe-information-bulk-200-response-inner-spec
  (ds/spec
    {:name ::get-recipe-information-bulk-200-response-inner
     :spec get-recipe-information-bulk-200-response-inner-data}))

File: angular/com/spoonacular/client/model/getRecipeInformationBulk200ResponseInner.ts
/**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over 380,000 recipes, thousands of ingredients, 800,000 food products, and 100,000 menu items. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what\'s in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { GetRecipeInformation200ResponseExtendedIngredientsInner } from './getRecipeInformation200ResponseExtendedIngredientsInner';
import { GetRecipeInformation200ResponseWinePairing } from './getRecipeInformation200ResponseWinePairing';


export interface GetRecipeInformationBulk200ResponseInner { 
    id: number;
    title: string;
    image: string;
    imageType: string;
    servings: number;
    readyInMinutes: number;
    license: string;
    sourceName: string;
    sourceUrl: string;
    spoonacularSourceUrl: string;
    aggregateLikes: number;
    healthScore: number;
    spoonacularScore: number;
    pricePerServing: number;
    analyzedInstructions: Array<string>;
    cheap: boolean;
    creditsText: string;
    cuisines: Array<string>;
    dairyFree: boolean;
    diets: Array<string>;
    gaps: string;
    glutenFree: boolean;
    instructions: string;
    ketogenic: boolean;
    lowFodmap: boolean;
    occasions: Array<string>;
    sustainable: boolean;
    vegan: boolean;
    vegetarian: boolean;
    veryHealthy: boolean;
    veryPopular: boolean;
    whole30: boolean;
    weightWatcherSmartPoints: number;
    dishTypes: Array<string>;
    extendedIngredients: Set<GetRecipeInformation200ResponseExtendedIngredientsInner>;
    summary: string;
    winePairing: GetRecipeInformation200ResponseWinePairing;
}

